File: Array/243_ShortestWordDistance.java
public class Solution {
    public int shortestDistance(String[] words, String word1, String word2) {
        int idx1 = -1, idx2 = -1;
        int n = words.length();
        int dis = Integer.MAX_VALUE;
        for (int i=0; i<n; i++) {
            if (words[i].equals(word1)) {
                if (idx2 != -1) dis = Math.min(dis, Math.abs(idx2 - idx1));
                idx1 = i;
            } else if (words[i].equals(word2)) {
                if (idx1 != -1) dis = Math.min(dis, Math.abs(idx2 - idx1));
                idx2 = i;
            }
        }
        return dis;
    }
}

File: 243_ShortestWordDistance.py
class Solution(object):
    def shortestDistance(self, words, word1, word2):
        """
        :type words: List[str]
        :type word1: str
        :type word2: str
        :rtype: int
        """
        pos1, pos2, dis = -1, -1, sys.maxint
        for i in range(len(words)):
            if words[i] == word1:
                pos1 = i
            if words[i] == word2:
                pos2 = i
            if not (pos1 == -1 or pos2 == -1):
                dis = min(abs(pos1 - pos2), dis)
        return dis